docs/design/server_side_charts.md

```diff
@@ -0,0 +1,44 @@
+* supporting the parameterization of charts, as opposed to templates.
+* supporting more than one top level template in a single chart.
+* supporting dependency definition and resolution mechanism at the chart level.
+
+It was also observed that there is strong motivation to complete the first pass of the helm/dm integration, using the existing DM templating engine, before adding new features or addressing more complex scenarios.
+
+## Assertions
+In view of this context, commentators asserted that:
+* It seems unlikely for a single chart to require more than one generation technology.
+* It seems unnecessary that we need to support more than one top level template per chart, if templates can nest, since any forest of top level templates required by the user can be implemented using a tree, where a single top level template calls the members of the forest.
+
+If we accept these assertions, then parameterizing a chart becomes the same thing as parameterizing a single top level template. This is an important result, because it means that:
+* we don't need to rationalize schemas, type systems or parameterization mechanisms across multiple technologies.
+* we don't need to worry about aggregating parameters from multiple templates in a chart level parameter block.
+* we don't need to examine the intermediate results of template expansion from one templating engine to see if they need further processing by a different one.
+* we can examine metadata, such as a `//helm:generate directive` in the top level template, or a property in the `Chart.yaml` file, to identify the templating engine for the top level template, and then hand that engine the chart, and let it chase template references and expand results all the way down to primitives.
+
+## Implications
+Note, however, that these assertions require templates to nest, which in turn requires a working dependency management mechanism that supports the definition and resolution of references. 
+
+Currently, DM templates provide such a mechanism. We resolve references iteratively during expansion (i.e., resolve references, expand, repeat). However, a reference resolves to a template file, which is assumed to be the only top level template file in the target folder. It may pull in other files from the folder, but it's the root of the file graph in that folder.
+
+If we preserve this model of a single top level template per folder, then the process of resolving template references doesn't have to change much to use charts instead of raw folders. References can resolve to charts, which we can retrieve to extract their contents (i.e,. since they may be tarballs or zipballs).
+
+If DM can pull files from charts, then there is no need for dependency management at the chart level, as long as we're working with DM templates. This is desirable, at least for the near term, because supporting dependency management at the chart level would require us to:
+* rework the DM dependency management mechanism, and
```

Comment:
No, if we put dependency resolution at the chart level and make it general, then DM will no longer be responsible for resolving its own references and expanding recursively.

IMO, doing that would be a lot of work, and a fools errand, since DM works well today, and since general dependency management that works well for arbitrary templating technologies would be hard to do well, if not impossible, given that we don't know the complete set of possible template types.